lib/web_ui/lib/src/engine/dom.dart

```diff
@@ -1739,27 +2807,29 @@ extension DomScreenExtension on DomScreen {
 class DomScreenOrientation extends DomEventTarget {}
 
 extension DomScreenOrientationExtension on DomScreenOrientation {
-  Future<dynamic> lock(String orientation) {
-    final Object jsResult = js_util.callMethod<Object>(this, 'lock', <String>[orientation]);
-    return js_util.promiseToFuture(jsResult);
-  }
-  external void unlock();
+  @JS('lock')
+  external JSPromise _lock(JSString orientation);
+  Future<dynamic> lock(String orientation) =>
+      js_util.promiseToFuture(_lock(orientation.toJS));
+
+  external JSVoid unlock();
```

Comment:
Why is `JSVoid` needed? I thought `void` would be enough? Can you do `unlock().toDart` ?

Reply:
lol, so `JSVoid` is currently just a typedef for `void`. But we all know that JS actually returns `undefined` in all cases. I haven't yet decided if we're going to support this. However, I'm going to make `JSVoid` an actual type that gets erased to `void` in the nearish future. Why? Because `void` is a Dart type, not a JS type. Technically some JS types are Dart types, but only because Dart doesn't give us any other option. Ohhh for a `Top` type someday.

I digress, JS has no notion of `void`. We provide `JSVoid` so users can signal their intent clearly. For the time being there is no `JSVoid.toDart`, nor `void.toJS`. The major benefit of not using `void` is that it allows us to completely disentangle JS types from Dart types, and it ensures that every single type on an external signature is a JS type.

I know this may seem painful now, but it is possible that the existence of `JSVoid` may enable optimization opportunities. For example, maybe it *is* more efficient to actually return `undefined` from JS. Is this true? I have no idea, but if it is, we would need `JSVoid`. Having it in place now ensures we're prepared for a possible future where it is valuable, and we can always revisit the decision down the road if we like. Personally, I think there is an nice symmetry to ensuring every type on an external function is a JS type, but ymmv.